ketch-android/Android Ketch Pref Center using Java/README.md

```diff
@@ -0,0 +1,198 @@
+        ...
+        
+        sharedPreferences = new KetchSharedPreferences(getApplicationContext());
+        
+        ...
+
+```
+
+Add the private member for storing the advertising ID and loading routines.
+
+[AAID](https://developer.android.com/training/articles/ad-id) is the suitable way to identify anonymous users,
+and we use it in this example for the sake of simplicity.
+
+There are other [ways to provide the user's identity information](https://docs.ketch.com/hc/en-us/articles/1500003453742-About-Identity-Spaces#field-descriptions-0-1).
+Use email or any other identifier that is available in you app.
+
+```java
+public class MainActivity extends AppCompatActivity {
+
+    ...
+
+    private interface AdvertisingIdCallback {
+        void onAdvertisingIdResult(@Nullable String advertisingId);
+    }
+
+    private void loadAdvertisingId(AdvertisingIdCallback callback) {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Handler handler = new Handler(Looper.getMainLooper());
+
+        progressBar.setVisibility(View.VISIBLE);
+
+        executor.execute(() -> {
+            String id = null;
+            try {
+                id = AdvertisingIdClient.getAdvertisingIdInfo(MainActivity.this).getId();
+            } catch (Exception e) {
+                Log.e(TAG, e.getMessage(), e);
+                handler.post(() -> {
+                    Toast.makeText(MainActivity.this,
+                            R.string.cannot_get_advertising_id_toast,
+                            Toast.LENGTH_LONG
+                    ).show();
+                });
+            } finally {
+                final String advertisingId = id;
+                handler.post(() -> {
+                    callback.onAdvertisingIdResult(advertisingId);
+                    progressBar.setVisibility(View.GONE);
+                });
+            }
+        });
+    }
+
+    ...
+
+}
+```
+
+### Step 4. Finally, setup the Ketch webview with the Ketch JS SDK configuration
+
+Create the webview class or use KetchWebView from this sample. Add ths view to your your main activity layout.
+In works in a similar way to The Ketch Smart Tag on the HTML page
+that expects an organization code and app property code to be passed in to it.
+
+Organization code `ORG_CODE` could be found on the [Organization Settings](https://app.ketch.com/settings/organization)
+
+App property code `PROPERTY` could be found on the [Properties Management](https://app.ketch.com/deployment/applications)
+
+Advertising ID code `ADVERTISING_ID_KEY` is available on the [Identity Spaces](https://app.ketch.com/settings/identity-spaces)
+
+The preferences screen is triggered on the button click in this example, but it could also be
+triggered automatically on application start.
+
+```java
+public class MainActivity extends AppCompatActivity {
+    
+    private static final String ORG_CODE = "transcenda";
+    private static final String PROPERTY = "website_smart_tag";
```

Comment:
Please update these to placeholder values.

Reply:
oh, I will fix